src/Plugin/PluginInterface.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * @copyright 2017 Hostnet B.V.
+ */
+declare(strict_types=1);
+
+namespace Hostnet\Component\Resolver\Plugin;
+
+/**
+ * Asset lib plugin interface. Any plugin should implement this interface.
+ *
+ * Example: adding a class that imports asset dependencies.
+ * ```
+ * $angular_collector = new AngularImportCollector();
+ * if ($plugin_api->getConfig()->isDev()) {
+ *     $angular_collector = new CachedImportCollector($angular_collector, $plugin_api->getCache());
+ * }
+ * $plugin_api->addCollector($angular_collector);
+ * ```
+ *
+ * Example: adding an event on post processing generated assets.
+ * ```
+ * $listener = new UglifyjsListener();
+ * $plugin_api->getConfig()->getEventDispatcher()->addListener(AssetEvents::POST_PROCESS, [$listener, 'onPostProcess']);
```

Comment:
Small formatting thing, perhaps the notation should be changed slightly to allow it on multiple lines like in the actual code in our plugins